Added pet renaming

ChargeIn opened the pull request:
Added support for renaming the pets.


Since testing might be a bit tricky since the scanbot logic is still missing you can use the attached patch file to allow deploying the scan bot of a scientist and rename it.


[scan_bot_deploy.patch](https://github.com/NexusForever/NexusForever/files/12346582/scan_bot_deploy.patch)

Rawaho commented:
It would be good if the rename validation could also be checked as part of this PR.
I assume this system already partially exists via UserText but would be good to check.

ChargeIn commented:
I added the same validation as used by the client. The only thing missing is the profanity filter. But since I'm not really familiar with the policy regarding external libs, I left a todo for now.

Rawaho commented:
This is actually already built into the core, see TextFilterManager and UserText.
It handles both text profanity and validity, it is based on how the client does it.

ChargeIn commented:
Ah nice, 👍.  I will be away for a week afterwards i will look into adding the missing logic.

Rawaho commented:
![image](https://github.com/NexusForever/NexusForever/assets/6111341/f69aee7c-6769-4c3a-a2e4-eae00b4aadf3)
When you get to it, these seem to be the values in the client.

ChargeIn commented:
Thx for the values. I tried to look into the client logic validation, but it seems a bit to much for me for now 😅.

I matched the values you provided, squashed my commits and cleaned up the branch.
